systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -557,12 +564,24 @@ protected void convertCRDs(JsonObject conversionTool, String namespace) throws I
 
         Exec.exec("chmod", "+x", convertorPath);
 
-        LOGGER.info("Converting CRs");
+        LOGGER.info("Converting CRs ...");
         // run conversion of crs
-        Exec.exec(convertorPath, "cr", "-n=" + namespace);
+        // CRs conversion may fail, because for 0.23 for example it's already done
+        // CRS conversion needs old versions of Strimzi CRDs, which are not available after 0.22
+        ExecResult execResult = Exec.exec(null, Arrays.asList(convertorPath, "cr", "-n=" + namespace), 0, true, false);
+        LOGGER.debug("CRs conversion STDOUT:");
+        LOGGER.debug(execResult.out());
+        LOGGER.debug("CRs conversion STDERR:");
+        LOGGER.debug(execResult.err());
+        LOGGER.info("CRs conversion done!");
         // run crd-upgrade
         LOGGER.info("Converting CRDs");
-        Exec.exec(convertorPath, "crd");
+        execResult = Exec.exec(convertorPath, "crd");
+        LOGGER.debug("CRDs conversion STDOUT:");
+        LOGGER.debug(execResult.out());
+        LOGGER.debug("CRDs conversion STDERR:");
+        LOGGER.debug(execResult.err());
+        LOGGER.info("CRDs conversion done!");
```

Comment:
I'm not sure I follow this part ... are we just ignoring if it fails?

Reply:
yes, I am open to any better solution, but in general, CRs conversion is not working in case that 0.23 is installed, because v1alpha1 is not instaled.

Reply:
I guess we can detect it what versions the CRDs have. Or maybe we should have some flag if it should be done or not? I guess we will remove the api-conversion tool now as well, so that would cause further problems.

Reply:
ok, detect CRD version make sense. I will change it.